From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TheKit <thekit@disroot.org>
Date: Tue, 16 Jun 2026 14:55:46 +0300
Subject: [PATCH] (halium) init.rc: skip wait for keystore.module_hash.sent

keystore2 is absent on Halium; this wait blocks boot forever.

Change-Id: I6f8391e7b36143619bbd7c1053f629a43d625799
---
 rootdir/init.rc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 8bfd941818..af6a41aae4 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -1079,7 +1079,9 @@ on post-fs-data
     wait_for_prop apexd.status activated
     # Wait for KeyMints to receive APEX module info before starting code from updateable APEXes.
     # This is to prevent APEX modules from interfering in module measurement.
-    wait_for_prop keystore.module_hash.sent true
+    # Halium: keystore2 is not present, so keystore.module_hash.sent is never set and this
+    # wait_for_prop blocks boot forever. Disabled to let boot proceed.
+    # wait_for_prop keystore.module_hash.sent true
     perform_apex_config
 
     start system_aconfigd_socket_service
